github.go

```diff
@@ -52,63 +55,74 @@ func runGitHubHandler(c *gin.Context) {
 		},
 	}
 
-	importer.Token, err = importer.getToken(i)
-	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to get token. %+v", err))
+	var ok bool
+	importer.Token, ok = importer.getTokenWritesProblem(c, i)
+	if !ok {
 		return
 	}
 
 	importer.Provider, err = importer.getProvider(i, importer.Token)
 	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to get provider. %+v", err))
+		ginutil.WriteAPIClientReadError(c, err,
+			fmt.Sprintf("Unable to get GitHub provider by ID %v or name %q", i.ProviderID, i.Provider))
 		return
 	}
 
 	importer.GithubClient, err = importer.initGithubConnection()
 	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to init github connection. %+v", err))
+		ginutil.WriteAPIClientReadError(c, err,
+			fmt.Sprintf("Unable to parse provider url %q or upload url %q",
+				importer.Provider.URL, importer.Provider.UploadURL))
 		return
 	}
 
 	if i.ProjectID != 0 || i.Project != "" {
 		err = importer.importProject(i)
 		if err != nil {
-			c.Error(err)
-			c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to import project. %+v", err))
+			ginutil.WriteAPIClientWriteError(c, err,
+				fmt.Sprintf("Unable to import project %q (id: %v) from GitHub.", i.Project, i.ProjectID))
 			return
 		}
 	} else {
 		err = importer.importGroup(i.Group)
 		if err != nil {
-			c.Error(err)
-			c.JSON(http.StatusBadRequest, fmt.Sprintf("Unable to import organization or group. %+v", err))
+			ginutil.WriteAPIClientWriteError(c, err,
+				fmt.Sprintf("Unable to import group %q from GitHub.", i.Group))
 			return
 		}
 	}
 
 	c.Status(http.StatusCreated)
 }
 
-func (importer githubImporter) getToken(i importBody) (wharfapi.Token, error) {
+func (importer githubImporter) getTokenWritesProblem(c *gin.Context, i importBody) (wharfapi.Token, bool) {
 	var token wharfapi.Token
 	var err error
 
 	if i.TokenID != 0 {
 		token, err = importer.WharfClient.GetTokenByID(i.TokenID)
 		if err != nil {
-			return token, err
+			ginutil.WriteAPIClientReadError(c, err,
+				fmt.Sprintf(
+					"Unable to get token by id %v. Likely because of a failed request or malformed response.",
+					i.TokenID))
 		} else if token.TokenID == 0 {
-			err = fmt.Errorf(fmt.Sprintf("Token with id %v not found", i.TokenID))
+			err = fmt.Errorf("token with id %v not found", i.TokenID)
+			ginutil.WriteAPIClientReadError(c, err,
+				fmt.Sprintf("Token with id %v not found", i.TokenID))
 		}
 	} else {
 		token, err = importer.WharfClient.PutToken(wharfapi.Token{Token: i.Token, UserName: i.User})
+		if err != nil {
+			ginutil.WriteAPIClientWriteError(c, err,
+				fmt.Sprintf(
+					"Unable to create token for %q. Likely because of a failed request or malformed response.",
```

Comment:
nit: Good to always describe the values, as values such as an empty string will not be self explanatory.

```suggestion
					"Unable to create token for user %q. Likely because of a failed request or malformed response.",
```